src/lib/hooks/usePostPreviewData.ts

```diff
@@ -0,0 +1,23 @@
+import { useSelector } from 'react-redux';
+import { IPost } from '../types';
+import { RootStateType } from '../../store/rootReducer';
+
+const usePostPreviewData = () => {
+  const { user } = useSelector((state: RootStateType) => state.currentUser);
+
+  const post = {
+    author: {
+      avatar: user?.avatar,
```

Comment:
Please add **id** field to the author - since it's required and can't be optional

Reply:
fixed with spread `user` to author obj.